nixos/tests/tigervnc.nix

```diff
@@ -0,0 +1,50 @@
+{ system ? builtins.currentSystem
+, config ? {}
+, pkgs ? import ../.. { inherit system config; }
+}:
+
+with import ../lib/testing-python.nix { inherit system pkgs; };
+
+let
+  makeVNCTest = name: vncPackage: (import ./make-test-python.nix ({
+    inherit name;
+    meta = with pkgs.stdenv.lib.maintainers; {
+      maintainers = [ iblech ];
+    };
+
+    nodes = {
+      server = { pkgs, ...}: {
+        environment.systemPackages = [ vncPackage pkgs.xorg.xwininfo pkgs.xteddy ];
+        networking.firewall.allowedTCPPorts = [ 5901 ];
+      };
+
+      client = { pkgs, ... }: {
+        imports = [ ./common/x11.nix ];
+        environment.systemPackages = [ vncPackage ];
+      };
+    };
+
+    testScript = ''
+      start_all()
+
+      for host in [server, client]:
+          host.succeed("mkdir /root/.vnc")
+          host.succeed("echo foobar | vncpasswd -f > /root/.vnc/passwd")
+
+      server.succeed("vncserver -geometry 1234x567 :1")
+      server.succeed("DISPLAY=:1 xwininfo -root | grep 1234x567")
+      server.execute("DISPLAY=:1 xteddy &")
+
+      client.wait_for_x()
+      client.execute("vncviewer server:1 &")
+      client.wait_for_window(r"VNC")
+      client.screenshot("screenshot")
+
+      server.succeed("vncserver -kill :1")
+    '';
+  }) { inherit system; });
+
+in with pkgs; {
+  testTigerVNC = makeVNCTest "tigervnc" tigervnc;
+  testTightVNC = makeVNCTest "tightvnc" tightvnc;
```

Comment:
Ok. Maybe the renaming was not the best idea. Just spotted that it tests two different implementations. However trying to build this, still causes infinite recursion.

Reply:
Yes, the idea was to have one test file for both packages, as they are very similar in their functionality.

What's the best way to proceed? If I'm honest then I'm not completely sure why we use `passthru.tests.tigervnc = nixosTests.tigervnc` which I believe to cause the infinite recursion. I loosely based the test on the test for Mumble, and the Mumble package also does not contain an analogue of this declaration.

Thank you tending to this pull request! :-)

Reply:
Actually I use the same thing here: https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/networking/ferm/default.nix#L30